spec/omni_auth/strategies/mlh_spec.rb

```diff
@@ -0,0 +1,196 @@
+        expect(result[:profile]).to eq({ age: 22, gender: 'Female' })
+      end
+    end
+
+    context 'with v4 API empty response' do
+      include_context 'with oauth response', {}
+
+      it 'returns an empty hash for empty data' do
+        expect(strategy.data).to eq({})
+      end
+    end
+
+    context 'with v4 API complex data structures' do
+      include_context 'with oauth response', {
+        'id' => 'test-id',
+        'education' => [{
+          'school' => {
+            'name' => 'Test University',
+            'location' => 'Test City'
+          },
+          'graduation_year' => 2024
+        }],
+        'social_profiles' => [
+          { 'platform' => 'github', 'url' => 'https://github.com' },
+          'https://twitter.com'
+        ],
+        'professional_experience' => [{
+          'company' => 'Tech Corp',
+          'positions' => [
+            { 'title' => 'Engineer', 'years' => [2022, 2023] }
+          ]
+        }]
+      }
+
+      it 'correctly processes complex nested structures' do
+        result = strategy.data
+        expect(result).to be_a(Hash)
+        expect(result[:education].first[:school]).to eq({ name: 'Test University', location: 'Test City' })
+        expect(result[:social_profiles]).to eq([{ platform: 'github', url: 'https://github.com' }, 'https://twitter.com'])
+        expect(result[:professional_experience].first[:positions].first[:years]).to eq([2022, 2023])
+      end
+    end
+
+    context 'with v4 API nil and empty values' do
+      include_context 'with oauth response', {
+        'id' => 'test-id',
+        'first_name' => 'Jane',
+        'last_name' => nil,
+        'profile' => {
+          'age' => 22,
+          'gender' => nil,
+          'location' => {
+            'city' => nil,
+            'country' => 'USA'
+          }
+        },
+        'education' => nil,
+        'social_profiles' => {
+          'github' => {},
+          'linkedin' => {}
+        }
+      }
+
+      it 'handles nil values correctly' do
+        result = strategy.data
+        expect(result[:last_name]).to be_nil
+        expect(result[:profile][:gender]).to be_nil
+        expect(result[:profile][:location]).to eq({ city: nil, country: 'USA' })
+        expect(result[:education]).to be_nil
+      end
+
+      it 'handles empty hash structures' do
+        result = strategy.data
+        expect(result[:social_profiles]).to eq({ github: {}, linkedin: {} })
+      end
+    end
+
+    context 'with API error' do
+      it 'returns empty hash on error' do
+        allow(access_token).to receive(:get).and_raise(StandardError)
+        expect(strategy.data).to eq({})
+      end
+    end
+  end
+
+  describe '#uid' do
+    context 'with valid data' do
+      before do
+        allow(strategy).to receive(:data).and_return({ id: 'test-123' })
+      end
+
+      it 'returns the id from the data hash' do
+        expect(strategy.uid).to eq('test-123')
+      end
```

Comment:
Since, we just have 1 test within the context, we can add the stub within the it block.